core/src/main/java/org/hibernate/ogm/persister/impl/SingleTableOgmEntityPersister.java

```diff
@@ -37,6 +38,10 @@ public SingleTableOgmEntityPersister(
 	private static EntityDiscriminator resolveDiscriminator(final PersistentClass persistentClass, final SessionFactoryImplementor factory) {
 		if ( persistentClass.isPolymorphic() ) {
 			Value discrimValue = persistentClass.getDiscriminator();
+			if ( discrimValue == null ) {
+				throw new MappingException( "discriminator mapping required for single table polymorphic persistence" );
```

Comment:
I copied this from Hibernate ORM, not sure how much we want to they similar to what they do